workshops/views.py

```diff
@@ -3154,10 +3147,132 @@ def duplicates(request):
 
     return render(request, 'workshops/duplicates.html', context)
 
+# ------------------------------------------------------------
+# Views for trainees
+
+def _homework2state(homework):
```

Comment:
This is utility function that is used only in `trainee_dashboard` view. It won't be used anywhere else, so why do you want to move it?